src/coreneuron/CMakeLists.txt

```diff
@@ -492,7 +492,7 @@ endforeach()
 
 # we can link to MPI libraries in non-dynamic-mpi build
 if(NRN_ENABLE_MPI AND NOT NRN_ENABLE_MPI_DYNAMIC)
-  target_link_libraries(coreneuron-core PUBLIC ${MPI_CXX_LIBRARIES})
+  target_link_libraries(coreneuron-core PUBLIC ${MPI_C_LIBRARIES})
```

Comment:
I seem to remember that it sometimes matters. When compiling as a C++ project, some implementations of MPI (I think OpenMPI on Ubuntu (or ArchLinux)) would rely on C++ functionality, even if we were only using the C API ourselves.

I think that's why we have:
https://github.com/BlueBrain/HighFive/blob/d51d6cc0c87737e5e2d68adfe297576161eae1a2/cmake/HighFiveConfig.cmake#L13-L16